rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -1024,6 +1025,29 @@ public ResponseEntity<CollectionModel<EntityModel<Project>>> getUsedByProjectDet
         return new ResponseEntity<>(attachmentUsageMap, status);
     }
 
+    @RequestMapping(value = PROJECTS_URL + "/import/SBOM", method = RequestMethod.POST, consumes = {MediaType.MULTIPART_FORM_DATA_VALUE})
+    public ResponseEntity<HalResource> importSBOM(@RequestParam(value = "type", required = true) String type,
+                                                  @RequestBody MultipartFile file) throws TException {
+        final User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+        Attachment attachment = null;
+        final RequestSummary requestSummary;
+        if(!type.equalsIgnoreCase("SPDX")) {
+            throw new IllegalArgumentException("SBOM file type is not valid. It currently only supports SPDX(.rdf/.xml) files.");
+        }
+        try {
+            attachment = attachmentService.uploadAttachment(file, new Attachment(), sw360User);
+            requestSummary = projectService.importSBOM(sw360User, attachment.getAttachmentContentId());
```

Comment:
- Move this line `projectService.importSBOM` outside `try-catch` block or add a nested `try-catch`, in case of any issues with import SBOM functionality, the error message would still be `failed to upload attachment` (which is wrong).
- Also check for `SUCCESS` requestStatus in `requestSummary` object (returned from importSBOM) and existence of `projectId` in `requestSummary.getMessage` before calling the API to get project information at line `1045`.
- correct the formatting of `if (!type.equalsIgnoreCase("SPDX")) {` clause at line `1034`.